src/main/java/org/broadinstitute/barclay/argparser/CommandLinePluginDescriptor.java

```diff
@@ -76,103 +110,127 @@
     public abstract List<String> getPackageNames();
 
     /**
-     * Give this descriptor a chance to filter out any classes it doesn't want to be
-     * dynamically discoverable.
-     * @return false if the class shouldn't be used; otherwise true
+     * Base class for all command line plugin classes managed by this descriptor. Subclasses of
+     * this class in any of the packages returned by {@link #getPackageNames} will be command line
+     * accessible.
      */
-    public Predicate<Class<?>> getClassFilter() { return c -> true;}
+    public abstract Class<?> getPluginBaseClass();
```

Comment:
Why not bounding this? In principle, the `CommandLinePluginDescriptor`should be for classes of `T`, no?

Reply:
In principle, we could add a lower bound (<? super T>) for this particular case, but for most of the rest of the cases, a lower bound makes the makes the method inaccessible to callers that don't have compile-time knowledge of T (like the command line parser), and an upper bound requires awkward constructs in the implementation. So, as I mentioned above, I'm advocating that we just leave all of the Class params unbounded to keep them consistent.

Reply:
Ok, I understand the problem now. But at least we should enforce the behavior in the contract by adding a `It must return a base class of type {@link T}` to the javadoc...